src/java/org/apache/cassandra/db/rows/Cell.java

```diff
@@ -42,8 +45,14 @@
 public abstract class Cell<V> extends ColumnData
 {
     public static final int NO_TTL = 0;
-    public static final int NO_DELETION_TIME = Integer.MAX_VALUE;
-    public static final int MAX_DELETION_TIME = Integer.MAX_VALUE - 1;
+    public static final long NO_DELETION_TIME = Long.MAX_VALUE;
+    // -5 to leave room for any future sentinel values just in case
```

Comment:
Do we have any idea about what those future values would be?

Reply:
No, but loosing 5s vs cornering yourself into a position you'll have no maneuvering room if you need it seems quite beneficial to me.

Reply:
You got me thinking that if we don't like this approach, when/if the time comes we need it we could use times in the past before the C* days as sentinel values. wdyt?

Reply:
In the latest commit i move it to -2. Future sentinels could be values in the past. Let me know what you think.

Reply:
So MAX - 2 is max deletion time, MAX - 1 is future use, and MAX is no deletion, right? If we keep a reserved future sentinel we should probably keep the previous comment, `-2 to leave room for any future sentinel values just in case`.

Reply:
MAX - 1 is `LEGACY_OVERFLOWED_DELETION_TIME` right?